File: src/net/merayen/elastic/ui/controller/ViewportController.java
package net.merayen.elastic.ui.controller;

import org.json.simple.JSONObject;

import net.merayen.elastic.system.intercom.NetListRefreshRequestMessage;
import net.merayen.elastic.ui.intercom.ViewportHelloMessage;
import net.merayen.elastic.ui.objects.top.viewport.ViewportContainer;
import net.merayen.elastic.util.Postmaster.Message;

public class ViewportController extends Controller {
	ViewportContainer viewport_container;

	public ViewportController(Gate gate) {
		super(gate);
	}

	@Override
	protected void onInit() {}

	@Override
	public void onMessageFromBackend(Message message) {
		if(message instanceof NetListRefreshRequestMessage) {
			
		}
	}

	@Override
	public void onMessageFromUI(Message message) {
		if(message instanceof ViewportHelloMessage) { // Received from ViewportContainer UIObject when it has inited. We can then manage it
			viewport_container = ((ViewportHelloMessage)message).viewport_container;
		}

		/*else if(message instanceof NetListRefreshRequestMessage) {
			sendToBackend(message);
		}*/
	}

	@Override
	protected JSONObject onDump() {
		return viewport_container.dump();
	}

	@Override
	protected void onRestore(JSONObject obj) {
		if(viewport_container == null)
			throw new RuntimeException("Should not happen");

		viewport_container.restore(obj);
	}
}

File: src/main/kotlin/net/merayen/elastic/ui/controller/ViewportController.kt
package net.merayen.elastic.ui.controller

import net.merayen.elastic.system.intercom.ElasticMessage
import net.merayen.elastic.ui.intercom.ViewportHelloMessage
import net.merayen.elastic.ui.objects.top.Top
import net.merayen.elastic.ui.objects.top.viewport.ViewportContainer

class ViewportController(top: Top) : Controller(top) {
	private var viewportContainer: ViewportContainer? = null

	override fun onInit() {}

	override fun onMessageFromBackend(message: ElasticMessage) {}

	override fun onMessageFromUI(message: ElasticMessage) {
		if (message is ViewportHelloMessage) // Received from ViewportContainer UIObject when it has inited. We can then manage it
			viewportContainer = message.viewport_container
	}
}
